infra/modules/bootstrap/main.tf

```diff
@@ -0,0 +1,252 @@
+data "aws_caller_identity" "current" {}
+
+data "aws_partition" "current" {}
+
+# Create the dynamodb table requried for state locking.
```

Comment:
We have 3 options here, an AWS owned Key, AWS managed key, or Customer managed key. I am not a fan of the AWS owned keys because they aren't unique to your account. Between the other two I figured, "might as well select the on one that gives you full control, just in case you need it. I like the phrasing that you have here a bit better, so I am gonna steal that.